docs/coding-conventions.md

```diff
@@ -0,0 +1,89 @@
+# Coding conventions
+
+This document outlines a collection of guidelines, style suggestions, and tips
+for writing code in the KubeVirt project. It is partially based on [Kubernetes
+Coding Conventions](https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md).
+
+- [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+- [Effective Go](https://golang.org/doc/effective_go.html)
+- Know and
+  avoid [Go landmines](https://gist.github.com/lavalamp/4bd23295a9f32706a48f)
+- Inline variable assignments when possible
+    - For example inline `err` checks
```

Comment:
I know I'm on the minority on this one, but I'll open it here to debate since maybe someone would agree :smile: 

It's a matter of style and taste I guess, but when I look at:
```go
err := functionThatDoesSomething(param1, param2, param3)
Expect(err).ToNot(HaveOccurred())
```

I can clearly see what's being executed and what's the validation being made.
While here:
```go
Expect(functionThatDoesSomething(param1, param2, param3)).To(Succeed())
```

It looks like a big mess. A long one-liner that kind of hides the fact that we do an operation + validation at the same time.

As said I know I'm the minority here and probably will lose this battle :)

Reply:
While in the specific `ginkgo` case I tend to agree with you, because you can additionally have the error descriptions in the `Expect` to further complicate the line, I can see the benefits of inlining the error in the following situations:

```
err := functionThatDoesSomething(param1, param2, param3)
if err != nil {
  return err
}
```

That could be rewritten as:

```
if err := functionThatDoesSomething(param1, param2, param3); err != nil
  return err
}
```

Reply:
@iholder101 Yes, I agree with that. It was meant for cases like @machadovilaca depicted.

@EdDev Does it really support functions returning multiple values (values+error)? From the docs it does not sound like that:

> You should not use a function with multiple return values (like DoSomethingHard) with Succeed

See https://onsi.github.io/gomega/#handling-errors